### PR TITLE
[main][feat]: add liquidity, token0Available and token1Available  at getRepur…

### PR DESCRIPTION
**Pull request by sundayalpaca**
…chaseSummary

## Description

add liquidity, token0Available and token1Available  at getRepurchaseSummary

### CODE REVIEW

File: src/reader/PancakeV3VaultReader.sol [hunk 1]
```diff
@@ -67,6 +69,7 @@ contract PancakeV3VaultReader is IVaultReader {
       _vaultSummary.tickUpper = _tickUpper;
       _vaultSummary.lowerPrice = _tickToPrice(_tickLower, _token0.decimals(), _token1.decimals());
       _vaultSummary.upperPrice = _tickToPrice(_tickUpper, _token0.decimals(), _token1.decimals());
+      _vaultSummary.poolSqrtPriceX96 = _poolSqrtPriceX96;
```
Comment (jr-alpaca) [inline src/reader/PancakeV3VaultReader.sol:72]: assign this outside if statement. Pool price doesn't depend on tokenId.

File: src/reader/PancakeV3VaultReader.sol [hunk 2]
```diff
@@ -100,20 +103,22 @@ contract PancakeV3VaultReader is IVaultReader {
...
         _getPositionFees(_tokenId, PancakeV3Worker(_worker).nftPositionManager(), PancakeV3Worker(_worker).pool());
 
       // TODO: include pending interest after upgrade mm
-      uint256 _token0PositionValue = (_vautlSummary.token0Undeployed + _vautlSummary.token0Farmed + _token0TradingFee)
-        * _vautlSummary.token0price / 1e18;
-      uint256 _token0DebtValue = _vautlSummary.token0Debt * _vautlSummary.token0price / 1e18;
-      uint256 _token1PositionValue = (_vautlSummary.token1Undeployed + _vautlSummary.token1Farmed + _token1TradingFee)
-        * _vautlSummary.token1price / 1e18;
-      uint256 _token1DebtValue = _vautlSummary.token1Debt * _vautlSummary.token1price / 1e18;
+      uint256 _token0PositionValue = (
```
Comment (jr-alpaca) [inline src/reader/PancakeV3VaultReader.sol:114]: please check why format change

File: src/reader/PancakeV3VaultReader.sol [hunk 3]
```diff
@@ -272,24 +280,29 @@ contract PancakeV3VaultReader is IVaultReader {
...
     } else {
-      _result.stableToken = address(PancakeV3Worker(_worker).token1());
-      _result.assetToken = address(PancakeV3Worker(_worker).token0());
+      _result.stableToken = _token1;
+      _result.assetToken = _token0;
       _result.stableTokenFees = _pendingRewards[1].amount;
       _result.assetTokenFees = _pendingRewards[0].amount;
     }
     _result.cakeFarmed = _pendingRewards[2].amount;
+    _result.token0Available = moneyMarket.getProtocolReserve(_token0);
```
Comment (alpanaca) [inline src/reader/PancakeV3VaultReader.sol:304]: It's not protocolReserve. Should use getFloatingBalance()

File: src/reader/PancakeV3VaultReader.sol [hunk 4]
```diff
@@ -255,6 +262,7 @@ contract PancakeV3VaultReader is IVaultReader {
     uint256 token1Debt;
     uint256 lowerPrice;
     uint256 upperPrice;
+    uint256 liquidity;
```
Comment (alpanaca) [inline src/reader/PancakeV3VaultReader.sol:265]: missing another param. poolAddress. Can get this from worker.pool()

File: src/interfaces/IVaultReader.sol [hunk 5]
```diff
@@ -16,6 +16,7 @@ interface IVaultReader {
     int24 tickLower;
     int24 tickUpper;
     uint128 liquidity;
+    uint256 poolSqrtPriceX96;
```
Comment (jr-alpaca) [inline src/interfaces/IVaultReader.sol:19]: wrong type
```suggestion
    uint160 poolSqrtPriceX96;
```

**Review by jr-alpaca — APPROVED** [review]

**Review by alpanaca — APPROVED** [review]

**Pull request merged**